### DISCUSSION TITLE
[Group Licensing 8.2.9] - Keep track of all licenses + Remove a license seat

### DISCUSSION THREAD

**Discussion by BenjaminYde** [dashingsoft/pyarmor#1377] [Q&A]
We got the pyarmor group license today.

- How do we keep track of all the used licenses of our group?
  - Some sort of `pyarmor reg --list`?
- Can we remove licenses in the group? How does this work? (make a license seat free again)
- When registering a device license with ID 1, is this ID then not usable on another device? Will an error pop-up?
- Let's say device with ID 1 is registered, can we later remove this device?
- It would be handy to see the status of all devices, when is the license generated / last used / device name / ...

Currently the only thing you can do is using `pyarmor reg ./pyarmor-device-regfile-XXXX.1.zip` to register a device ID

**Comment by jondy** [comment]
Once ID 1 is assigned to one device, it can be used again. It's impossible to remove seat ID 1, then reuse it again.

Since it's not removable, so checking the device files in the `.pyarmor/group` could get all the devices.
```
ls .pyarmor/group/pyarmor-group-device.*
```

Now only machine id in the `pyarmor-group-device.*`, in next release, `platform.uname()`, computer name or any other useful information will be included as comment lines.

> Currently the only thing you can do is using pyarmor reg ./pyarmor-device-regfile-XXXX.1.zip to register a device ID

You can also generate `pyarmor-group-device.2` and then `pyarmor-device-regfile-xxxx.2.zip`, and `.3`, `.4` to `.100`

**Reply by BenjaminYde** [reply]
So it is recommended to keep an excel that contains a list of all the ID's with the attached PC name etc.

**Comment by jondy** [comment]
Keep all the `Pyarmor-group-device.N` in one folder, use a simple awk script to query all the device informations.
